integration-tests/src/test/java/tachyon/client/FileOutStreamAsyncWriteIntegrationTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client;
+
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import tachyon.TachyonURI;
+import tachyon.client.file.FileOutStream;
+import tachyon.client.file.TachyonFile;
+import tachyon.client.file.options.OutStreamOptions;
+import tachyon.heartbeat.HeartbeatContext;
+import tachyon.heartbeat.HeartbeatScheduler;
+import tachyon.master.file.meta.PersistenceState;
+import tachyon.thrift.FileInfo;
+import tachyon.util.CommonUtils;
+import tachyon.util.io.PathUtils;
+
+/**
+ * Integration tests for {@link tachyon.client.file.FileOutStream} of under storage type being async
+ * persist.
+ *
+ */
+public final class FileOutStreamAsyncWriteIntegrationTest extends FileOutStreamIntegrationTest {
```

Comment:
I see. So the subclassing integration tests only inherit basic member variables from the parent integration tests? Then in that case, I am not very sure about the benefit of having this integration test a subclass. For me the downside is, it will be tricky to maintain the parent integration tests---if I want to modify the parent integration tests, I need to be very careful what I could modify as it may break the subclassing integration test.